tsdb/querier.go

```diff
@@ -380,6 +389,10 @@ func inversePostingsForMatcher(ix IndexReader, m *labels.Matcher) (index.Posting
 			}
 		}
 		for _, val := range vals {
+			if ctx.Err() != nil {
```

Comment:
Same as the other loop over `vals`, I think this will be overkill.

Reply:
Perhaps `vals` can be batched and the context checked before each batch is started/

Reply:
it make sense!